jsonnet/config.libsonnet

```diff
@@ -0,0 +1,258 @@
+{
+  // import kube-prometheus and override configurations
+  kp:
+    (import 'kube-prometheus/main.libsonnet') +
+    // Uncomment the following imports to enable its patches
+    // (import 'kube-prometheus/addons/anti-affinity.libsonnet') +
+    // (import 'kube-prometheus/addons/managed-cluster.libsonnet') +
+    (import 'kube-prometheus/addons/node-ports.libsonnet') +
+    // (import 'kube-prometheus/addons/static-etcd.libsonnet') +
+    // (import 'kube-prometheus/addons/custom-metrics.libsonnet') +
+    // (import 'kube-prometheus/addons/external-metrics.libsonnet') +
+    {
+      values+:: {
+        common+: {
+          namespace: $._config.openebsMonitoring.namespace,
+        },
+      },
+    },
+
+  // import openebs-mixin and override configurations
+  openebsMixin: (import './openebs-mixin/mixin.libsonnet') {
+    _config+:: {
+      dashboards+: {
+        cStor: $._config.casType.cStor.dashboards,
+        Jiva: $._config.casType.Jiva.dashboards,
+        LocalPV: $._config.casType.lvmLocalPV.dashboards,
+        lvmLocalPV: $._config.casType.lvmLocalPV.dashboards,
+        deviceLocalPV: $._config.casType.lvmLocalPV.dashboards,
+      },
+      alertRules+: {
+        cStor: $._config.casType.cStor.alertRules,
+        Jiva: $._config.casType.Jiva.alertRules,
+        lvmLocalPV: $._config.casType.lvmLocalPV.alertRules,
+        deviceLocalPV: $._config.casType.lvmLocalPV.alertRules,
+      },
+    },
+  },
+
+  // Template function for ServiceMonitor
+  local serviceMonitor(port, path, selectorName) = {
+    // Endpoints of the selected service to be monitored
+    endpoints: {
+
+      // Name of the endpoint's service port
+      port: port,
+
+      // HTTP path to scrape for metrics
+      path: path,
+    },
+    // Label selector for services to which this ServiceMonitor applies
+    selector: {
+      matchLabels: selectorName,
+    },
+    // Namespaces from which services are selected
+    namespaceSelector: {
+      any: true,
+    },
+  },
+
+  // Template function for PodMonitor
+  local podMonitor(port, path, selectorName) = {
```

Comment:
where are we using podMonitors?

Reply:
For cStor pool pods, we are creating a PodMonitor.